test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,1002 @@
+		const altValue = await page.getByLabel( imageAltLabel ).inputValue();
+		expect( altValue ).toBe( textCustomFieldValue );
+
+		// Title input is enabled and with the original value.
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+		await expect( page.getByLabel( imageTitleLabel ) ).toBeEnabled();
+		const titleValue = await page
+			.getByLabel( imageTitleLabel )
+			.inputValue();
+		expect( titleValue ).toBe( 't' );
+	} );
+
+	test( 'Image - should disable title input when title is bound', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( titleBindingImageBlock );
+		const imageBlockImg = editor.canvas
+			.getByRole( 'document', {
+				name: 'Block: Image',
+			} )
+			.locator( 'img' );
+		await imageBlockImg.click();
+
+		// Replace controls exist.
+		await expect(
+			page.getByRole( 'button', {
+				name: imageReplaceName,
+			} )
+		).toBeVisible();
+
+		// Image src is the placeholder.
+		const imageSrc = await imageBlockImg.getAttribute( 'src' );
+		expect( imageSrc ).toBe( imagePlaceholder );
+
+		// Alt textarea is enabled and with the original value.
+		await expect( page.getByLabel( imageAltLabel ) ).toBeEnabled();
+		const altValue = await page.getByLabel( imageAltLabel ).inputValue();
+		expect( altValue ).toBe( 'a' );
+
+		// Title input is disabled and with the custom field value.
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+		await expect( page.getByLabel( imageTitleLabel ) ).toBeDisabled();
+		const titleValue = await page
+			.getByLabel( imageTitleLabel )
+			.inputValue();
+		expect( titleValue ).toBe( textCustomFieldValue );
+	} );
+
+	test( 'Image - multiple bindings should lock the appropriate controls', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( multipleBindingsImageBlock );
+		const imageBlockImg = editor.canvas
+			.getByRole( 'document', {
+				name: 'Block: Image',
+			} )
+			.locator( 'img' );
+		await imageBlockImg.click();
+
+		// Replace controls don't exist.
+		await expect(
+			page.getByRole( 'button', {
+				name: imageReplaceName,
+			} )
+		).toBeHidden();
+
+		// Image src is the custom field value.
+		const imageSrc = await imageBlockImg.getAttribute( 'src' );
+		expect( imageSrc ).toBe( urlCustomFieldValue );
+
+		// Alt textarea is disabled and with the custom field value.
+		await expect( page.getByLabel( imageAltLabel ) ).toBeDisabled();
+		const altValue = await page.getByLabel( imageAltLabel ).inputValue();
+		expect( altValue ).toBe( textCustomFieldValue );
+
+		// Title input is enabled and with the original value.
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+		await expect( page.getByLabel( imageTitleLabel ) ).toBeEnabled();
+		const titleValue = await page
+			.getByLabel( imageTitleLabel )
+			.inputValue();
+		expect( titleValue ).toBe( 't' );
+	} );
+} );
+
+test.describe( 'Block bindings - Template context', () => {
```

Comment:
What's really the difference between the template and the post context? These tests look mostly the same.

Reply:
Not too much, I've remove the ones I considered repetitive.